docs/standard/native-interop/tutorial-comwrappers.md

```diff
@@ -0,0 +1,936 @@
+
+```csharp
+var cw = new DemoComWrappers();
+var demo = new DemoImpl();
+IntPtr ccw = cw.GetOrCreateComInterfaceForObject(demo, CreateComInterfaceFlags.None);
+```
+
+### Step 3 &ndash; Implementing `CreateObject()`
+
+Construction of Native Object Wrappers has more implementation options and a great deal more nuance. The first question to address is permissive the `ComWrappers` subclass will be in support COM types. If the idea is to support all COM types, which is possible, a substantial amount of code will need to be written and/or some very clever uses of `Reflection.Emit` will need to be employed. For this tutorial we will only be support COM instances that implement both `IDemoGetType` and `IDemoStoreType`. Since we know there is a finite set and have restricted that any supplied COM instance must implement both interfaces we could provide a single statically defined wrapper; however, dynamic cases are common enough in COM we will explore both options.
+
+#### Static Native Object Wrapper
+
+We will look at the static implementation first. The static Native Object Wrapper involves defining a managed type that implements the .NET interfaces and can then forward the calls on the managed type to the COM instance. A rough outline of the static wrapper is below.
+
+```csharp
+class DemoNativeStaticWrapper
+    : IDemoGetType
+    , IDemoStoreType
+{
+    public string? GetString() =>
+        throw new NotImplementedException();
+
+    public void StoreString(int len, string? str) =>
+        throw new NotImplementedException();
+}
+```
+
+Constructing an instance of this class and providing it as a wrapper requires defining some policy. If this type is used as a wrapper it would seem that since it implements both interfaces that the underlying COM instance should implement both interfaces too. Given that we adopting this policy we need to ensure this through calls to [`Marshal.QueryInterface()`][api_marshalqueryinterface] on the COM instance to confirm.
+
+```csharp
+int hr = Marshal.QueryInterface(ptr, ref IDemoGetType.IID_IDemoGetType, out IntPtr IDemoGetTypeInst);
+if (hr != 0)
+{
+    return null;
+}
+
+hr = Marshal.QueryInterface(ptr, ref IDemoStoreType.IID_IDemoStoreType, out IntPtr IDemoStoreTypeInst);
+if (hr != 0)
+{
+    Marshal.Release(ptr);
+    return null;
+}
+
+return new DemoNativeStaticWrapper()
+{
+    IDemoGetTypeInst = IDemoGetTypeInst,
+    IDemoStoreTypeInst = IDemoStoreTypeInst
+};
+```
+
+#### Dynamic Native Object Wrapper
+
+Dynamic wrappers are more flexible in that they provide a way for types to be query at run time instead of statically. In order to provide this support we will utilize [`IDynamicInterfaceCastable`][api_idynamicinterfacecastable]. Observe that `DemoNativeDynamicWrapper` only implements this interface. The functionality that the interface provides is a chance at run time to determine what type is supported.
```

Comment:
```suggestion
Dynamic wrappers are more flexible becaues they provide a way for types to be queried at run time instead of statically. In order to provide this support, you'll utilize [`IDynamicInterfaceCastable`][api_idynamicinterfacecastable]. Observe that `DemoNativeDynamicWrapper` only implements this interface. The functionality that the interface provides is a chance to determine what type is supported at run time.
```